### PR TITLE
Update governance docs around code of conduct violations

### PR DESCRIPTION
**Pull request by SgtCoDFish**
It was recommended in our graduation review (see #35) that we:

> ...add a process for removing Maintainers (and SC members) for reasons other than inactivity, such as violating the CoC or disruptive behavior.

This commit attempts to codify that removal process.

Most of the legwork for the process is already done in the CNCF [foundation repo](https://github.com/cncf/foundation/tree/main/code-of-conduct).

Note that this expands the scope of the steering committee slightly by adding a new requirement, and this PR obviously changes our governance process. As such, this PR will go through a lazy consensus process and steering committee members will be explicitly asked to review.

### CODE REVIEW

**Review by PT-GD — APPROVED** [review]
Looks good from my perspective as an "external stakeholder" and contributor

File: CODE_OF_CONDUCT.md
Comment (erikgb) [inline CODE_OF_CONDUCT.md:16]:
```suggestion
that team shall be made up of both the steering committee and maintainers.
```

Or do we actually need to include all maintainers?

File: GOVERNANCE.md
Comment (erikgb) [inline GOVERNANCE.md:311]: No chance of ever becoming a member again? Really) 😉

File: GOVERNANCE.md
Comment (erikgb) [inline GOVERNANCE.md:310]:
```suggestion
appeals, that person will be removed and will not be eligible to rejoin the
```

**Review by erikgb — COMMENTED** [review]
Thanks for putting together this, Ashley! I've added a couple of comments/suggestions.

**Comment by FlorianLiebhart** [comment]
👍 Looks good from my side as well! :)

File: GOVERNANCE.md
Comment (SgtCoDFish) [reply GOVERNANCE.md:311]: Honestly I couldn't imagine a scenario where someone would get kicked out of the project and then rejoin it! Maybe if I added like a year's delay and they they could start again? But I just can't imagine how we'd ever get someone doing that!

File: CODE_OF_CONDUCT.md
Comment (SgtCoDFish) [reply CODE_OF_CONDUCT.md:16]: Given how rare this should be I think it's reasonable to ask all maintainers to attend - but that said, I think you're right that it's probably too strict to say "all" here since some might not have the time to spare.

I'll rewrite it to say everyone is asked to join, and to take into account that not everyone is required.

**Comment by SgtCoDFish** [comment]
All reviews welcome on this! Thank you to everyone who's already reviewed 🚀 

Unless there's a blocker, the [lazy consensus period](https://kubernetes.slack.com/archives/CDEQJ0Q8M/p1725624632531529) for reviewing will expire on 2024-09-12 at 12:00 London time and this will be accepted

File: GOVERNANCE.md
Comment (PT-GD) [inline GOVERNANCE.md:311]: is this one year timeline a CNCF-derived requirement? I can both conceive of minor incidents where there's a clear "get-well" plan such as "provide evidence of taking a remedial training, craft an appropriate apology to the community, recommit to the code of conduct" where a "weeks to months" outcome-based timeline was appropriate.

File: GOVERNANCE.md
Comment (PT-GD) [inline GOVERNANCE.md:311]: disregard previous comment...I see that this section is *specifically* when a a removal action is the result of an incident response, and doesn't preclude other responses

File: CODE_OF_CONDUCT.md
Comment (munnerz) [inline CODE_OF_CONDUCT.md:23]: Does the CNCF provide any other level of CoCC that we could refer these matters to? I'm conscious if say, a contributor/non-maintainer has an issue with a maintainer or steering committee members behaviour, they may be discouraged from making the initial report in the knowledge that the alleged perpetrator would be involved in/aware of the report.

File: GOVERNANCE.md
Comment (munnerz) [inline GOVERNANCE.md:317]: I understand the motivation/rationale for this, however, it may be contentious to take this action *immediately*...

I also wonder if this could be an issue wrt. privacy too - if a regular community member reports another regular community member, would we inform the accused party *immediately* that there is an investigation into them underway, or would we withhold this until an appropriate time/when required? With this policy in place, once a member with some form of privileges has a complaint filed against them, they'd be immediately aware due to permissions changes.

Comment (munnerz) [reply GOVERNANCE.md:317]: I wonder if we should also explicitly list what is in-scope and out-of-scope for these permission changes too.. would we remove the member from GitHub orgs (to prevent them being able to edit/delete comments, manage labels on issues & PRs?).

I think the intent of these changes overall are fairly clear, but the devil is in the detail a bit, and it's particularly sensitive when it comes to code of conduct violations as these are usually, by definition, quite personal reports.

File: CODE_OF_CONDUCT.md
Comment (SgtCoDFish) [reply CODE_OF_CONDUCT.md:23]: Fair enough yeah; no reason not to be explicit. I've added a link to the relevant section and more prose.

File: GOVERNANCE.md
Comment (SgtCoDFish) [reply GOVERNANCE.md]: I've tried to expand on this with more links to the CNCF documentation which I think states this well. Ultimately we'll always defer to the CNCF on this stuff.

**Comment by SgtCoDFish** [comment]
Thanks for the review @munnerz  - I've made some minor updates which I hope should address your comments!

File: CODE_OF_CONDUCT.md
Comment (munnerz) [reply CODE_OF_CONDUCT.md:23]: Those changes look great - way clearer and acknowledges up front that there may be some cases where people feel uncomfortable to report to all maintainers at once. Thank you! 🙌

**Review by munnerz — APPROVED** [review]
This looks great, thanks @SgtCoDFish!

**Comment by SgtCoDFish** [comment]
Thanks for the review + hold! I'll wait until lazy consensus ends on 2024-09-12 at 12:00 London time before merging, just to give everyone the best chance to review (there's no rush I don't think!)

**Comment by SgtCoDFish** [comment]
(The changes are just me rebasing against the latest commits on `main` - no text changes)

**Review by erikgb — APPROVED** [review]
/lgtm

Looks very nice! Thanks, Ash!

**Comment by SgtCoDFish** [comment]
/unhold

Lazy consensus has ended and there aren't any blockers so this is accepted and will be merged. Thanks to everyone that participated!

**Pull request merged**
